src/modules/cxxpiper.cpp

```diff
@@ -0,0 +1,1223 @@
+
```

Comment:
This needs to get a header providing the licence under which you ship the module. Apparently you have copy/pasted a lot of code from piper, so it *has* to respect the licensing, so copy/paste the content of LICENSE.md from there.

Reply:
This is really needed.